src/Compilers/CSharp/Portable/BoundTree/BoundExpression.cs

```diff
@@ -55,6 +55,11 @@ internal bool NeedsToBeConverted()
                 case BoundKind.TupleLiteral:
                 case BoundKind.UnconvertedSwitchExpression:
                     return true;
+                case BoundKind.StackAllocArrayCreation:
+                    // A BoundStackAllocArrayCreation is given a null type when it is in a
+                    // syntactic context where it could be either a pointer or a span, and
+                    // in that case it requires conversion to one or the other.
```

Comment:
Please capture this information in the BoundNodes.xml above the Type property definition.